### PR TITLE
Use `RegisterDirectiveOrder` for `rate_limit` before `basic_auth`

### PR DESCRIPTION
**Pull request by steffenbusch**
Fixes https://github.com/mholt/caddy-ratelimit/issues/58

Changes
* Updated the `init` function to include `RegisterDirectiveOrder`.
* Modified `go.mod` to upgrade the requirement from `github.com/caddyserver/caddy/v2 v2.7.6` to `v2.8.0`.
* Updated the README to reflect the new default order of the `rate_limit` directive.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,11 +111,12 @@ Sweep interval configures how often to scan for expired rate limiters. The defau
 
 ### Caddyfile config
 
-As with all non-standard HTTP handler modules, this directive is not known to the Caddyfile adapter and so it must be ["ordered"](https://caddyserver.com/docs/caddyfile/directives#directive-order) manually using global options unless it only appears within a [`route` block](https://caddyserver.com/docs/caddyfile/directives/route). This ordering usually works well, but you should use discretion:
+By default, the `rate_limit` directive is ordered before `basic_auth` in the Caddyfile. This simplifies configuration and removes the need for manual ordering in most cases. 
+However, if this default order does not suit your needs, you can still change the order using the [`order`](https://caddyserver.com/docs/caddyfile/directives#directive-order) global directive. For example:
```
Comment (mholt) [inline README.md:115]: We should restore mention of the `route` directive since it can be useful too.

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks -- we can try this; I think I agree with the placement, I just have one nit about the readme.

File: README.md (hunk 1 above)
Comment (steffenbusch) [reply README.md:115]: I have restored the mention of `route` and added an example for this as well.

**Review by mholt — APPROVED** [review]
Thanks very much!

**Pull request merged**